doc/tutorial/alignment_file/alignment_file_read_cigar.cpp

```diff
@@ -0,0 +1,44 @@
+#include <fstream>
+
+#include <seqan3/std/filesystem>
+
+struct write_file_dummy_struct
+{
+    write_file_dummy_struct()
+    {
+
+auto file_raw = R"//![sam_file](
+@HD	VN:1.6	SO:coordinate
+@SQ	SN:ref	LN:45
+r001	99	ref	7	30	8M2I4M1D3M	=	37	39	TTAGATAAAGGATACTG	*
+r003	0	ref	9	30	5S6M	*	0	0	GCCTAAGCTAA	*	SA:Z:ref,29,-,6H5M,17,0;
+r004	0	ref	16	30	6M14N5M	*	0	0	ATAGCTTCAGC	*
+r003	2064	ref	29	17	5M	*	0	0	TAGGC	*	SA:Z:ref,9,+,5S6M,30,1;
+r001	147	ref	37	30	9M	=	7	-39	CAGCGGCAT	*	NM:i:1
+)//![sam_file]";
+
+        std::ofstream file{std::filesystem::temp_directory_path()/"my.sam"};
+        std::string str{file_raw};
+        file << str.substr(1); // skip first newline
+    }
+};
+
+write_file_dummy_struct go{};
+
+//![code]
```

Comment:
well this code is completely copy and pastable, it just excludes the file creation above so I guess a snippet is still fine?

Reply:
Well, the reason is that the snippet when executed will create a file called with a generic name in a generic location maybe used by some other snippets as well. So it would be possible that if they get executed they overwrite something another process is using. That is when for example we start to explicitly check for the right output of the files in tests. But also it writes something into the tmp directory and not removing it anymore. So we decided to make all snippets work with streams if they are not explicitly demonstrating some file interface. I guess that is the case here. All I/O snippets were adapted to this design.

Reply:
again, can't we use the tmp_filename ?

Reply:
yes we can, but we have to port it to core or something.

Reply:
Can I leave it like this and then port the tmp file ?